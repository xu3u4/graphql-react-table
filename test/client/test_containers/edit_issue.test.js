import React from 'react';
import { Provider } from 'react-redux';
import { mount, shallow } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';

import InitialState from 'reducers';
import EditIssueContainer from 'containers/edit_issue';

describe('<EditIssue> container', () => {
  const middleWares = [thunk];
  const mockStore = configureMockStore(middleWares);
  let action = () => ({}), wrapper, store;

  beforeEach(() => {
    store = mockStore(InitialState(undefined, action()));
    store.dispatch(action());
    wrapper = shallow(
      <EditIssueContainer store={store}/>
    );
  });
  describe('with default settings', () => {
    it('should render <EditIssue>', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('dispatch an action', () => {
    action = () => ({ type: 'UPDATE_ISSUE_SUCCESS' });
    it('should get dispatched action', () => {
      const getAction = store.getActions();
      expect(getAction).toEqual([{ type: 'UPDATE_ISSUE_SUCCESS' }]);
    });
  });
});
