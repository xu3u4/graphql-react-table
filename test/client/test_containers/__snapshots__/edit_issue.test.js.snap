// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditIssue> container with default settings should render <EditIssue> 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <EditIssue
    columns={
        Array [
            Object {
              "key": "seq",
              "label": "seq",
            },
            Object {
              "key": "Status",
              "label": "Status",
            },
            Object {
              "key": "Category",
              "label": "Category",
            },
            Object {
              "key": "Title",
              "label": "Title",
            },
            Object {
              "key": "Owner",
              "label": "Owner",
            },
            Object {
              "key": "Priority",
              "label": "Priority",
            },
            Object {
              "key": "Action",
              "label": "Action",
            },
          ]
    }
    createIssue={[Function]}
    isShowWarning={false}
    selectedIssue={Object {}}
    showWarning={[Function]}
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
    storeSubscription={
        Subscription {
            "listeners": Object {
              "notify": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": null,
          }
    }
    updateIssues={[Function]}
/>,
  "nodes": Array [
    <EditIssue
      columns={
            Array [
                  Object {
                    "key": "seq",
                    "label": "seq",
                  },
                  Object {
                    "key": "Status",
                    "label": "Status",
                  },
                  Object {
                    "key": "Category",
                    "label": "Category",
                  },
                  Object {
                    "key": "Title",
                    "label": "Title",
                  },
                  Object {
                    "key": "Owner",
                    "label": "Owner",
                  },
                  Object {
                    "key": "Priority",
                    "label": "Priority",
                  },
                  Object {
                    "key": "Action",
                    "label": "Action",
                  },
                ]
      }
      createIssue={[Function]}
      isShowWarning={false}
      selectedIssue={Object {}}
      showWarning={[Function]}
      store={
            Object {
                  "clearActions": [Function],
                  "dispatch": [Function],
                  "getActions": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                }
      }
      storeSubscription={
            Subscription {
                  "listeners": Object {
                    "notify": [Function],
                  },
                  "onStateChange": [Function],
                  "parentSub": undefined,
                  "store": Object {
                    "clearActions": [Function],
                    "dispatch": [Function],
                    "getActions": [Function],
                    "getState": [Function],
                    "replaceReducer": [Function],
                    "subscribe": [Function],
                  },
                  "unsubscribe": null,
                }
      }
      updateIssues={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Connect(EditIssue)
        store={
                Object {
                        "clearActions": [Function],
                        "dispatch": [Function],
                        "getActions": [Function],
                        "getState": [Function],
                        "replaceReducer": [Function],
                        "subscribe": [Function],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Connect {
        "_reactInternalInstance": [Circular],
        "context": Object {
          "store": undefined,
          "storeSubscription": undefined,
        },
        "notifyNestedSubs": [Function],
        "props": Object {
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "propsMode": true,
        "refs": Object {},
        "renderCount": 0,
        "selector": Object {
          "error": null,
          "props": Object {
            "columns": Array [
              Object {
                "key": "seq",
                "label": "seq",
              },
              Object {
                "key": "Status",
                "label": "Status",
              },
              Object {
                "key": "Category",
                "label": "Category",
              },
              Object {
                "key": "Title",
                "label": "Title",
              },
              Object {
                "key": "Owner",
                "label": "Owner",
              },
              Object {
                "key": "Priority",
                "label": "Priority",
              },
              Object {
                "key": "Action",
                "label": "Action",
              },
            ],
            "createIssue": [Function],
            "isShowWarning": false,
            "selectedIssue": Object {},
            "showWarning": [Function],
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "updateIssues": [Function],
          },
          "run": [Function],
          "shouldComponentUpdate": false,
        },
        "setWrappedInstance": [Function],
        "state": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "subscription": Subscription {
          "listeners": Object {
            "notify": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": null,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
        "version": 0,
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <EditIssue
          columns={
                    Array [
                              Object {
                                "key": "seq",
                                "label": "seq",
                              },
                              Object {
                                "key": "Status",
                                "label": "Status",
                              },
                              Object {
                                "key": "Category",
                                "label": "Category",
                              },
                              Object {
                                "key": "Title",
                                "label": "Title",
                              },
                              Object {
                                "key": "Owner",
                                "label": "Owner",
                              },
                              Object {
                                "key": "Priority",
                                "label": "Priority",
                              },
                              Object {
                                "key": "Action",
                                "label": "Action",
                              },
                            ]
          }
          createIssue={[Function]}
          isShowWarning={false}
          selectedIssue={Object {}}
          showWarning={[Function]}
          store={
                    Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "clearActions": [Function],
                                "dispatch": [Function],
                                "getActions": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": null,
                            }
          }
          updateIssues={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <EditIssue
          columns={
                    Array [
                              Object {
                                "key": "seq",
                                "label": "seq",
                              },
                              Object {
                                "key": "Status",
                                "label": "Status",
                              },
                              Object {
                                "key": "Category",
                                "label": "Category",
                              },
                              Object {
                                "key": "Title",
                                "label": "Title",
                              },
                              Object {
                                "key": "Owner",
                                "label": "Owner",
                              },
                              Object {
                                "key": "Priority",
                                "label": "Priority",
                              },
                              Object {
                                "key": "Action",
                                "label": "Action",
                              },
                            ]
          }
          createIssue={[Function]}
          isShowWarning={false}
          selectedIssue={Object {}}
          showWarning={[Function]}
          store={
                    Object {
                              "clearActions": [Function],
                              "dispatch": [Function],
                              "getActions": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                            }
          }
          storeSubscription={
                    Subscription {
                              "listeners": Object {
                                "notify": [Function],
                              },
                              "onStateChange": [Function],
                              "parentSub": undefined,
                              "store": Object {
                                "clearActions": [Function],
                                "dispatch": [Function],
                                "getActions": [Function],
                                "getState": [Function],
                                "replaceReducer": [Function],
                                "subscribe": [Function],
                              },
                              "unsubscribe": null,
                            }
          }
          updateIssues={[Function]}
/>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Connect(EditIssue)
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
}
`;
